### PR TITLE
Go server updates

### PR DESCRIPTION
**Pull request by adreyfus-stripe**
* Used FormValue
* Renamed method
* Handled Stripe error
* Ran gofmt

Thanks! 
@andybons

### CODE REVIEW

File: server/go/server.go [hunk 1]
```diff
@@ -98,26 +98,33 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
 
 	// Fetch the CheckoutSession object from your success page
 	// to get details about the order
-	session, _ := session.Get(
-		keys[0],
+	session, err := session.Get(
+		id,
```
Comment (andybons) [inline server/go/server.go]: Put this all on one line?

File: server/go/server.go [hunk 2]
```diff
@@ -98,26 +98,33 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
...
 		nil,
 	)
 
+	if err != nil {
+		log.Println("An error happened when getting the CheckoutSession from Stripe", id)
```
Comment (andybons) [inline server/go/server.go]: Print the error as well?

**Review by andybons — APPROVED** [review]
Looks great with some minor comments. Thanks for doing this.

File: server/go/server.go [hunk 3]
```diff
@@ -98,25 +98,29 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
...
 
-func handleGetCheckoutSession(w http.ResponseWriter, r *http.Request) {
+func handleCheckoutSession(w http.ResponseWriter, r *http.Request) {
 	if r.Method != "GET" {
 		http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
 		return
 	}
 
-	keys, ok := r.URL.Query()["sessionId"]
+	id := r.FormValue("sessionId")
```
Comment (andybons) [inline server/go/server.go:108]:
```suggestion
```

File: server/go/server.go [hunk 4]
```diff
@@ -98,25 +98,29 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
...
+	}
 
 	// Fetch the CheckoutSession object from your success page
 	// to get details about the order
-	session, _ := session.Get(
-		keys[0],
-		nil,
-	)
+	session, err := session.Get(id, nil)
+
```
Comment (andybons) [inline server/go/server.go:118]:
```suggestion
```

File: server/go/server.go [hunk 5]
```diff
@@ -98,25 +98,29 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		log.Println("An error happened when getting the CheckoutSession "+id+" from Stripe:", err)
```
Comment (andybons) [inline server/go/server.go]:
```suggestion
		log.Printf("An error happened when getting the CheckoutSession %q from Stripe: %v", id, err)
```

**Review by andybons — APPROVED** [review]

**Comment by adreyfus-stripe** [comment]
Thanks again @andybons ! 👏

**Comment by andybons** [comment]
Yay thanks, @adreyfus-stripe! Hope my lack of context in the suggestions or comments came off poorly. Sometimes I forget to say _why_ I'm suggesting changes.

**Comment by adreyfus-stripe** [comment]
Not at all @andybons! I appreciate attention to detail especially since I'm brand new to Go and still learning best practices.

**Pull request merged**